nemo/collections/multimodal/modules/stable_diffusion/quantization_utils/utils.py

```diff
@@ -0,0 +1,130 @@
+        "timesteps": {0: "steps"},
+        "y": {0: "batch_size"},
+        "context": {0: "batch_size", 1: "sequence_length"},
+    }
+}
+
+
+def generate_dummy_inputs(sd_version, device, latent_dim=32, adm_in_channels=1280):
+    dummy_input = {}
+    if sd_version == "nemo":
+        dummy_input["x"] = torch.ones(2, 4, latent_dim, latent_dim).to(
+            device
+        )  ## latent dim should be sampling resolution // 8
+        dummy_input["y"] = torch.ones(2, adm_in_channels).to(
+            device
+        )  ## adm_in_channels is 1280 when conditioner in consisted of only tokenizers, otherwise each additional embedding adds 512 in this dimention
+        dummy_input["timesteps"] = torch.ones(2).to(device)
+        dummy_input["context"] = torch.ones(2, 80, 2048).to(device)
+    else:
+        raise NotImplementedError(f"Unsupported sd_version: {sd_version}")
+
+    return dummy_input
+
+
+def load_calib_prompts(batch_size, calib_data_path="./calib_prompts.txt"):
+    with open(calib_data_path, "r") as file:
+        lst = [line.rstrip("\n") for line in file]
+    return [lst[i : i + batch_size] for i in range(0, len(lst), batch_size)]
+
+
+def get_int8_config(model, quant_level=3, alpha=0.8, percentile=1.0, num_inference_steps=20, global_min=False):
+    quant_config = {
+        "quant_cfg": {
+            "*lm_head*": {"enable": False},
+            "*output_layer*": {"enable": False},
+            "default": {"num_bits": 8, "axis": None},
+        },
+        "algorithm": {"method": "smoothquant", "alpha": alpha},
+    }
+    for name, module in model.named_modules():
+        w_name = f"{name}*weight_quantizer"
+        i_name = f"{name}*input_quantizer"
+
+        if w_name in quant_config["quant_cfg"].keys() or i_name in quant_config["quant_cfg"].keys():
+            continue
+        if filter_func(name):
+            continue
+        if isinstance(module, (torch.nn.Linear, LinearWrapper)):
+            if (
+                (quant_level >= 2 and "ff.net" in name)
+                or (quant_level >= 2.5 and ("to_q" in name or "to_k" in name or "to_v" in name))
+                or quant_level == 3
+            ):
+                quant_config["quant_cfg"][w_name] = {"num_bits": 8, "axis": 0}
+                quant_config["quant_cfg"][i_name] = {"num_bits": 8, "axis": -1}
+        elif isinstance(module, torch.nn.Conv2d):
+            quant_config["quant_cfg"][w_name] = {"num_bits": 8, "axis": 0}
+            quant_config["quant_cfg"][i_name] = {
+                "num_bits": 8,
+                "axis": None,
+                "calibrator": (
+                    PercentileCalibrator,
+                    (),
+                    {
+                        "num_bits": 8,
+                        "axis": None,
+                        "percentile": percentile,
+                        "total_step": num_inference_steps,
+                        "global_min": global_min,
```

Comment:
We changed `global_min` in `quant_config` in the latest AMMO diffusers, but it should be fine here as long as we verified the image quality. @jingyu-ml to confirm.

Reply:
I think the image quality is good, though I checked latest main and didn't get what is changed here. I might be missing something?

Reply:
We just merged an MR in AMMO gitlab main branch that updated `quant_config`, but I think we are good to go here as long as the image quality is good.